partiql-app/partiql-cli/src/main/kotlin/org/partiql/cli/InputSource.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.cli
+
+import java.io.File
+import java.io.InputStream
+
+internal sealed class InputSource {
+
+    internal abstract fun stream(): InputStream
+
+    internal class FileStreamSource(val file: File) : InputSource() {
+        override fun stream(): InputStream = file.inputStream()
+    }
+
+    internal class InputStreamSource(private val stream: InputStream) : InputSource() {
+        override fun stream(): InputStream = stream
+    }
+}
```

Comment:
The way I got to understand `InputSource`, it is a stream that can be re-opened multiple times while (like any stream) hiding away the source of data (a file or whatever).  I think this can become a pretty neat abstraction, so here are a few thoughts on how it could be improved.  

First, it does not quite fulfill the above idea: while `FileStreamSource` is "re-openable" (at each call to the `stream()` method, `InputStreamSource` is not.  However, `InputStreamSource` is currently instantiated only once (at CliTestUtility:30), from a string.  So, I think it can be re-done to follow `FileStreamSource`, except instantiating from a string rather than a file -- therefore becoming re-openable as well.  
Then better names for both would be `FileSource` and `StringSource`, I'd suggest. 

Second, the abstraction is "leaky" currently.  Ideally, one would create an `InputSource` (from a file or a string, or whatever) and then only use methods like `stream()` that do not reveal anything about the underlying source.  That would be a "tight" abstraction we are striving for.  However, in `Cli.runWithIonInputWrapped` it discriminates on the underlying source and, in the file-based case even gets to the underlying file.  There is also some duplication of code there and in `IonFileConverterIterable`.  
It is my hope that it can be refactored, probably moving some of the duplicated code to somewhere here in `InputSource`.  The goal is to have in `runWithIonInputWrapped` a simple call like `input.stream()` -- with no discrimination on the underlying input there, in `runWithIonInputWrapped`.
I admit I did not think through all of the details, so let me know if there is an obstacle that I overlooked of if it is not clear what I am suggesting.